src/cairoUtilFuncGen/memory/arrayLiteral.ts

```diff
@@ -67,16 +96,26 @@ export class MemoryArrayLiteralGen extends StringIndexedFuncGen {
 
     const argString = mapRange(size, (n) => `e${n}: ${elementCairoType.toString()}`).join(', ');
 
+    // If it's dynamic we need to include the length at the start
+    const alloc_len = dynamic ? size * elementCairoType.width + 2 : size * elementCairoType.width;
     this.generatedFunctions.set(key, {
       name: funcName,
       code: [
         `func ${funcName}{range_check_ptr, warp_memory: DictAccess*}(${argString}) -> (loc: felt):`,
         `    alloc_locals`,
-        `    let (start) = wm_alloc(${uint256(size * elementCairoType.width)})`,
-        mapRange(size, (n) => elementCairoType.serialiseMembers(`e${n}`))
-          .flat()
-          .map((name, index) => `dict_write{dict_ptr=warp_memory}(${add('start', index)}, ${name})`)
-          .join('\n'),
+        `    let (start) = wm_alloc(${uint256(alloc_len)})`,
+        [
+          ...(dynamic ? [`wm_write_256{dict_ptr=warp_memory}(start, ${uint256(size)})`] : []),
+          ...mapRange(size, (n) => elementCairoType.serialiseMembers(`e${n}`))
+            .flat()
+            .map(
+              (name, index) =>
+                `dict_write{dict_ptr=warp_memory}(${add(
+                  'start',
+                  dynamic ? index + 2 : index,
```

Comment:
Just curious, why is index increased by two when it is a dynamic array

Reply:
The length is stored as a Uint256 at the start of the memory area